src/tango_sdp_subarray/SDPSubarray/SDPSubarray.py

```diff
@@ -198,52 +251,76 @@ def ReleaseResources(self, config=''):
+    def Configure(self, pb_config):
         """Configure the device to execute a real-time Processing Block (PB).
 
         Provides PB configuration and parameters needed to execute the first
         scan in the form of a JSON string.
 
         :param pb_config: JSON string wth Processing Block configuration.
-        :param schema_path: Path to the PB config schema (optional).
         """
-        # pylint: disable=unused-argument
-        self._obs_state = ObsState.CONFIGURING
-        # time.sleep(1)
+        self._require_obs_state([ObsState.IDLE])
 
-        # Validate the SBI config schema
-        if schema_path is None:
-            schema_path = join(dirname(__file__), 'schema',
-                               'configure_pb.json')
+        LOG.info('Command: Configure (%s)', self.get_name())
+
+        LOG.debug('Setting ObsState to CONFIGURING.')
+        self._obs_state = ObsState.CONFIGURING
 
+        LOG.debug('Validating PB configuration.')
+        try:
+            pb_config = json.loads(pb_config)
+        except json.JSONDecodeError as error:
+            LOG.error('Unable to load JSON PB configuration: %s', error.msg)
+            raise
+        schema_path = join(dirname(__file__), 'schema', 'configure_pb.json')
         with open(schema_path, 'r') as file:
             schema = json.loads(file.read())
-
-        pb_config = json.loads(pb_config)
-        validate(pb_config, schema)
-
-        for txn in self.db_client.txn():
-            confdata = pb_config['configure']
-            pb = entity.ProcessingBlock(confdata['id'], None,
-                                        confdata['workflow'],
-                                        confdata['parameters'],
-                                        confdata['scanParameters'])
-            txn.create_processing_block(pb)
-
+        try:
+            validate(pb_config, schema)
+        except exceptions.ValidationError as error:
+            LOG.error('PB configuration failed to validate: %s', error.message)
+            frame_info = getframeinfo(currentframe())
+            error_message = '{} : {} : {}'.format(
+                str(error.absolute_path),
+                str(error.schema_path),
+                error.message
+            )
+            Except.throw_exception(
+                'Command: Configure failed', error_message,
+                '{}:{}'.format(frame_info.filename, frame_info.lineno))
+
+        # Add the PB configuration to the database.
+        if HAVE_CONFIG_DB and self._toggle_config_db:
+            for txn in self.db_client.txn():
+                conf_data = pb_config['configure']
+                pb = entity.ProcessingBlock(conf_data['id'],
+                                            None,
+                                            conf_data['workflow'],
+                                            conf_data['parameters'],
+                                            conf_data['scanParameters'])
```

Comment:
This might actually raise something if validation fails. Not sure about that, our approach to validation is likely a bit fragmented right now.